mesonbuild/coredata.py

```diff
@@ -190,27 +191,34 @@ def printable_value(self) -> str:
     def validate_value(self, value: T.Any) -> T.Union[str, OctalInt]:
         if value is None or value == 'preserve':
             return 'preserve'
-        return OctalInt(super().validate_value(value))
+        if isinstance(value, str):
```

Comment:
I haven't investigated for myself, but what's the reason we can't inherit from integer options anymore and have to copy this over?

Reply:
Because in that case we end up with conflicting types for `_T`, `UserIntegerOption` requires that `_T is int` while `UserOption[str | OctalInt]` requires `_T in [str, OctalInt]`, so you end up with things like `validate_value` must simultaneously return an `int` and `str | OctalInt`, which is obviously impossible.

Reply:
Yay... and I guess using a mixin or `UserIntegerOption.validate_value(self, value)` feel too hacky or something?

Reply:
I didn't think about using a mixin... that would work